### PR TITLE
skeleton for assert_response

### PR DESCRIPTION
**Pull request by ttung**
This allows us to build simple test statements.  The next diff in this stack will refactor the existing tests to use the new facility.

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -16,6 +21,48 @@
...
     def setUp(self):
         self.app = dss.create_app().app.test_client()
+        self.sre = re.compile("^assert(.+)Response")
+
+    def assertResponse(
+            self,
+            method: str,
+            path: str,
+            expected_code: int,
+            expected_headers: typing.Dict={},
```
Comment (kislyuk) [inline tests/test.py]: for my education, why do you say `typing.Dict` instead of `dict` here?

File: tests/test.py [hunk 2]
```diff
@@ -16,6 +21,48 @@
...
+            json_response_tester: typing.Optional[typing.Callable[[typing.Any], bool]]=None,
```
Comment (kislyuk) [inline tests/test.py]: I would suggest using this pattern instead:
```
def assert_response(..., expect_json=True, **kwargs):
    if "body" in kwargs and isinstance(kwargs["body"], dict):
        headers["content-type"] = "application/json"
        kwargs["body"] = json.dumps(kwargs["body"])
    if expect_json:
        return res.json() # or whatever means we must use to unpack json for now
    else:
        return res
```
This avoids boilerplate and allows me to chain assertions directly or tersely on the return value.

Alternatively, we could do another wrapper around this helper to implement this (e.g. assert_json_response).

File: tests/test.py (hunk 1 above)
Comment (ttung) [inline tests/test.py]: Will revise.

File: tests/test.py (hunk 2 above)
Comment (ttung) [inline tests/test.py]: Adopted aspects of your suggested changes.

File: tests/test.py [hunk 3]
```diff
@@ -16,6 +21,62 @@
...
+        self.sre = re.compile("^assert(.+)Response")
+
+    def assertResponse(
+            self,
+            method: str,
+            path: str,
+            expected_code: int,
+            json_request_body: typing.Optional[dict]=None,
+            parse_response_as_json: bool=False,
+            **kwargs) -> typing.Tuple[wrappers.Response, str, typing.Optional[dict]]:
```
Comment (kislyuk) [inline tests/test.py]: Another typing newbie comment: can I just use `(x, y)` instead of `Tuple[x, y]`?

**Comment by kislyuk** [comment]
Is there a missing `wrappers` import? Tests are failing.

File: tests/test.py (hunk 3 above)
Comment (ttung) [inline tests/test.py]: Doesn't seem like it.

**Comment by ttung** [comment]
Moved import from #36 
Removed the `parse_response_as_json` parameter (instead just try-except it)

**Pull request merged**